### ISSUE TITLE
select() has different behavior for "not" ! vs -

### ISSUE THREAD

**Issue by coltongearhart** [hadley/r4ds#1528]
```r
diamonds %>% select(1:5, !2) # doesn't result in what is expected
diamonds %>% select(1:5, -2) # does result in what is expected
```

Not sure if this is by design or not (pun intended)

**Comment by mine-cetinkaya-rundel** [comment]
Reported in the linked issue above, will address depending on feedback there.

**Comment by hadley** [comment]
Yeah they have different behaviour in this case. What do we say in the book?

**Comment by mine-cetinkaya-rundel** [comment]
We say:

> You can also use `-` instead of `!` (and you’re likely to see that in the wild); we recommend `!` because it reads as “not”, and combines well with `&` and `|`.

Here is what I suggest:

``` r
library(tidyverse)
library(nycflights13)

# book example - no match
flights |> select(!year:day) |> ncol()
#> [1] 16
flights |> select(-year:day) |> ncol()
#> Warning in x:y: numerical expression has 18 elements: only the first used
#> [1] 2

# suggested update to book - match * update chapter to include parens in the example
flights |> select(!(year:day)) |>ncol()
#> [1] 16
flights |> select(-(year:day)) |>ncol()
#> [1] 16

# issue example - no match
diamonds |> select(1:5, !2) |> ncol()
#> [1] 10
diamonds |> select(1:5, -2) |> ncol()
#> [1] 4

# suggested update to issue example - match
diamonds |> select(1:5 & !2) |> ncol()
#> [1] 4
diamonds |> select(1:5 & -2) |> ncol()
#> [1] 4
```

<sup>Created on 2023-11-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mine-cetinkaya-rundel** [comment]
Addressed with #1598.
